packages/core/admin/admin/src/components/GuidedTour/GuidedTourModal/index.js

```diff
@@ -34,15 +30,24 @@ const GuidedTourModal = () => {
 
     const [isStepDone] = at(guidedTourState, currentStep);
 
-    setIsVisible(!isStepDone);
-  }, [currentStep, guidedTourState]);
+    setIsVisible(!isStepDone && isGuidedTourVisible);
+  }, [currentStep, guidedTourState, isGuidedTourVisible]);
 
   useEffect(() => {
     if (currentStep) {
       const [content] = at(layout, currentStep);
+      const sectionKeys = Object.keys(guidedTourState);
+      const [sectionName, stepName] = currentStep.split('.');
+      const newSectionIndex = sectionKeys.indexOf(sectionName);
+      const newStepIndex = Object.keys(guidedTourState[sectionName]).indexOf(stepName);
+      const newHasSectionAfter = newSectionIndex < sectionKeys.length - 1;
+
       setStepContent(content);
+      setSectionIndex(newSectionIndex);
+      setStepIndex(newStepIndex);
+      setHasSectionAfter(newHasSectionAfter);
```

Comment:
There are too much setState here. IMO you should use a reducer function and dispatch a single action here.